### PR TITLE
Forms upload files

### PR DESCRIPTION
**Pull request by MarekBodingerBA**
Commit by commit.

This is a first MVP version that can be shipped. 🚢 

There are many things to be done:
 - Add tests for most of the use cases of the service.
 - Implement error messages properly (this also needs a team alignment).
 - The summary doesn't display file errors nor file upload status yet.
 - The form state doesn't know about file errors and doesn't prevent submission (this is a part of form state refactor).
 - Filesize is not displayed (the server doesn't return it yet).
 - UploadedFile shows only the text, not the visuals (will be probably handled by @radoslavzeman).
 - Disabled file upload for non-auth users.
 - Trigger files removal on form save.
 - Supported file types / size in environmental variables.
 - Display all the supported file types in upload widgets.
 - Move form loader logic to the server (including files).

### CODE REVIEW

**Review by mpinter — APPROVED** [review]
🙌 very nice

have only read through though,  but nothing unseemly popped up on me

I'd consider not using the await-to-js lib (see the one comment below), the one place it is used here (if I missed other), it could easily be a try-catch block as well

I count on you keeping track of the state of the followups - if there are any which you can define well as a separate tasks, would be nice to do so, so that @janickoR (or other dev) can pick some of them up and the workload can be split

**Review by radoslavzeman — COMMENTED** [review]
Consider shortening of name prefix from `FormFileUpload` to `FileUpload` as it is quit long and harder to read. Ad we won't have other "File Upload"  and everything is "Form"  :D

File: next/package.json [hunk 1]
```diff
@@ -40,24 +40,24 @@
     "ajv": "^8.11.0",
     "ajv-formats": "^2.1.1",
     "amazon-cognito-identity-js": "^6.2.0",
+    "await-to-js": "^3.0.0",
```
[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply next/package.json]: 👍

File: next/package.json [hunk 2]
```diff
@@ -40,24 +40,24 @@
...
     "libphonenumber-js": "^1.10.21",
     "libxmljs2": "^0.31.0",
     "lodash": "4.17.21",
-    "minio": "^7.0.32",
     "next": "13.4.5",
     "next-i18next": "12.1.0",
     "next-query-params": "^4.1.0",
     "react": "18.2.0",
-    "react-aria": "3.25.0",
+    "react-aria": "3.26.0",
```
[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply next/package.json:58]: 👍

**Pull request merged**